isy-security/src/main/java/de/bund/bva/isyfact/security/oauth2/client/Authentifizierungsmanager.java

```diff
@@ -2,58 +2,120 @@
+ * <p>
+ * The preferred way is to use {@link #authentifiziere(String)} which will automatically
  */
 public interface Authentifizierungsmanager {
 
     /**
-     * Attempts to authorize the client for the given {@code registrationId} via its configured OAuth 2.0 Flow.
-     * After successful authentication the authenticated principal in the {@link org.springframework.security.core.context.SecurityContext}
-     * will be updated.
+     * Attempts to authorize the client for the given {@code oauth2ClientRegistrationId} via its configured OAuth 2.0 Flow.
+     * After successful authentication the authenticated principal in the {@link SecurityContext} will be updated.
+     * <p>
      * The chosen OAuth 2.0 Flow will depend on the authorization grant type configured in the application properties.
      * The currently supported flows are:
      * <ul>
-     *     <li>Client Credentials Flow (grant type: client_credentials)</li>
-     *     <li>Resource Owner Password Credentials Flow (grant type: password)</li>
+     *     <li>Client Credentials (grant type: client_credentials)</li>
+     *     <li>Resource Owner Password Credentials (grant type: password)</li>
      * </ul>
+     * Additional properties for the Resource Owner Password Credentials Flow will be resolved via the matching
+     * registration in {@link IsyOAuth2ClientConfigurationProperties#registration}.
      *
-     * @param registrationId
-     *         ID of the OAuth 2.0 Client to authorize
+     * @param oauth2ClientRegistrationId
+     *         registration ID of the OAuth 2.0 Client to authorize
      * @throws AuthenticationException
      *         if authentication fails
      */
-    void authentifiziere(String registrationId) throws AuthenticationException;
+    void authentifiziere(String oauth2ClientRegistrationId) throws AuthenticationException;
 
     /**
      * Attempts to create and authorize a client with the given credentials via the OAuth 2.0 Client Credentials Flow.
-     * After successful authentication the authenticated principal in the {@link org.springframework.security.core.context.SecurityContext}
-     * will be updated.
+     * After successful authentication the authenticated principal in the {@link SecurityContext} will be updated.
      * <p>
-     * This method is only intended to ease migration to the Client Credentials Flow.
-     * If possible it is preferred to use {@link #authentifiziere(String)} instead.
+     * This method is only intended to allow OAuth 2.0 Client Authentication in cases where the credentials
+     * are obtained externally and cannot be configured in the application properties.
+     * If possible it is strongly preferred to use {@link #authentifiziere(String) authentication with a client registration ID} instead.
      *
+     * @param issuerLocation
+     *         Issuer used to query the discovery endpoints and set the token endpoint for authentication
```

Comment:
That seems to be all very correct. But aren't that too much internal details about _spring_ will internally act to connect at the end the _IAM system_?
As a user of _IsyFact_ especially of `isy-security`, these internal details should be hided and is one of the reasons to use such a library to simplify by transparency. Otherwise the application development could use _spring_ directly.

So on my opinion the decription of the parameter `issuerLocation` should only give the info, that it will contain the URI of the _IAM service_, which will be used for authentication.

Reply:
no changes necessary, because the details abaout the discovery endpoint are important.